### PR TITLE
Adding Fav Polling

### PR DESCRIPTION
**Pull request by gauntface**
This adds polling to the dashboard (updated every 30 minutes). This will change the favicon if there is a new actionable item.

### CODE REVIEW

File: src/client/public/scripts/dash/dash.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
-import {html, render} from '../../../../node_modules/lit-html/lit-html.js';
-import * as api from '../../../types/api';
+import {html, render} from '../../../../../node_modules/lit-html/lit-html.js';
+import * as api from '../../../../types/api';
+import { PullRequest } from '../../../../types/api';
```
Comment (samuelli) [inline src/client/public/scripts/dash/dash.ts]: nit: formatter

File: src/client/public/scripts/dash/dash.ts [hunk 2]
```diff
@@ -179,23 +199,113 @@ function prTemplate(pr: api.PullRequest) {
...
+    workLock = true;
+  
+    try {
+      await performPollAction();
+    } catch (err) {
+      console.log('Unable to perform poll update: ', err);
+    }
+  
+    workLock = false;
+  }, POLLING_INTERVAL);
```
Comment (samuelli) [inline src/client/public/scripts/dash/dash.ts]: Using `setTimeout` might avoid the need for the workLock? Not entirely convinced the lock is needed.

File: src/client/public/scripts/dash/dash.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
-import {html, render} from '../../../../node_modules/lit-html/lit-html.js';
-import * as api from '../../../types/api';
+import {html, render} from '../../../../../node_modules/lit-html/lit-html.js';
+import * as api from '../../../../types/api';
+import { PullRequest } from '../../../../types/api';
+
+// Poll every 30 Minutes
+const POLLING_INTERVAL = 30 * 60 * 1000;
```
Comment (samuelli) [inline src/client/public/scripts/dash/dash.ts]: 30 seems really long. The dashboard needs to be pretty timely, so I would prefer if this was significantly shorter (in the order of minutes).

Comment (gauntface) [reply src/client/public/scripts/dash/dash.ts]: 5 Minutes? I just don't want to basically kill a users quota.

Comment (samuelli) [reply src/client/public/scripts/dash/dash.ts]: 5 is good for now. If this becomes problematic, we can look into reducing our query costs.

**Pull request merged**